### PR TITLE
[4.0] Workflow Transitions/Stages

### PR DESCRIPTION
**Pull request by brianteeman**
1. Change the editor field to a text field - I am sure you will never need a full wysiwyg editor here
2. Add a label to the field called "Note"
3. change the tmpl to get the list of fields from the xml instead of being hardcoded
4. Add the content of the "note" to the list view. I really can't see the point in having the note if you can't see it in the list

(Same has been done for stages)

### After
chrome_2018-08-13_17-23-24
chrome_2018-08-13_17-23-34

### CODE REVIEW

File: administrator/components/com_workflow/tmpl/transition/edit.php [hunk 1]
```diff
@@ -35,23 +35,9 @@
...
-					<div class="control-label">
-						<?php echo $this->form->getLabel('to_stage_id'); ?>
-					</div>
-					<div class="controls">
-						<?php echo $this->form->getInput('to_stage_id'); ?>
-					</div>
-				</div>
-				<?php echo $this->form->getInput('description'); ?>
+			<?php foreach ($this->form->getGroup('transition') as $field) : ?>
+						<?php echo $field->renderField(); ?>
```
Comment (wilsonge) [inline administrator/components/com_workflow/tmpl/transition/edit.php]: You've added some extra tabs here

File: administrator/components/com_workflow/forms/transition.xml [hunk 2]
```diff
@@ -1,9 +1,10 @@
 <?xml version="1.0" encoding="utf-8"?>
 <form>
+	<fields name="transition">
```
Comment (wilsonge) [inline administrator/components/com_workflow/forms/transition.xml]: Can we indent the fieldset here if we're squeezing in the `<fields>` element please

**Review by wilsonge — APPROVED** [review]

**Review by wilsonge — APPROVED** [review]

**Comment by wilsonge** [comment]
LGTM. Let's just get one test here

**Comment by brianteeman** [comment]
Moved the name above the tabs for consistency

**Comment by brianteeman** [comment]
Something isnt correct here - probably a silly error by me - but the edit is not displaying the note or saving correctly. I will take another look in the morning unless some kind sole wants to submit a pr to my branch with the fix

**Comment by wilsonge** [comment]
You've added the fields tag. So the fields names have changed to e.g. `jform[transition][description]` instead of `jform[description]` before this PR. So you'll need to update the populateState function I would imagine https://github.com/joomla/joomla-cms/blob/51ea31b5e2c3791a8e4de968499add5746a4b8c7/administrator/components/com_workflow/Model/TransitionModel.php#L39-L48 

(just doing this off code review - please check what i've written before attempting anything)

**Comment by brianteeman** [comment]
Makes sense. Will check in the morning

**Comment by brianteeman** [comment]
And probably the getState

**Comment by brianteeman** [comment]
OK now updated for stages and ready for testing

**Comment by brianteeman** [comment]
@bembelimen I think this is clearer in the edit view and also means you can see all the notes at once in the list view - hope you agree

**Comment by bembelimen** [comment]
In general I really like this PR, but two things:
- perhaps move the note field to the right?
- What is the reason for grouping the fields into the "transitions fields"? Will this be extended in some way, so we need the foreach in the tmpl? I would suggest to use fieldset only, so we have not this nested field naming George mentioned.

**Comment by brianteeman** [comment]
> perhaps move the note field to the right?

I assume you mean in the list view? It is below to be consistent with similar additional information on other lists - also allows a bit more space and if it was to wrap onto a second line it will look better this way

**Comment by wilsonge** [comment]
> What is the reason for grouping the fields into the "transitions fields"? Will this be extended in some way, so we need the foreach in the tmpl? I would suggest to use fieldset only, so we have not this nested field naming George mentioned.

We don't have nested fields anymore with this (that's when you use a "fields" tag) this just names the fieldsets, which is required for allowing plugins to add more custom fields (for example see what I've done with the groups view in https://github.com/joomla/joomla-cms/pull/21696)

**Comment by bembelimen** [comment]
No I mean in the form view.

grafik

**Comment by bembelimen** [comment]
@wilsonge But there is a difference: you add the name to the fieldset, here we added a new "fields" tag: https://github.com/joomla/joomla-cms/pull/21592/files#diff-6d3a9c00497a5503546ac43bbad8023eR10

**Comment by brianteeman** [comment]
1. Note - i think thats a just a matter of opinion - I dont care and am happy to move it
2. Fields - my fault - didnt save the change - have pushed now

**Comment by brianteeman** [comment]
conflicts resolved

File: administrator/components/com_workflow/tmpl/stages/default.php [hunk 3]
```diff
@@ -122,8 +122,10 @@
 											<a href="<?php echo $edit; ?>" title="<?php echo Text::_('JACTION_EDIT'); ?> <?php echo $this->escape(addslashes($item->title)); ?>">
 												<?php echo $editIcon; ?><?php echo $item->title; ?>
 											</a>
+											<div class="small"><?php echo $item->description; ?></div>
 										<?php else: ?>
 											<?php echo $item->title; ?>
+											<div class="small"><?php echo $item->description; ?></div>
```
Comment (bembelimen) [inline administrator/components/com_workflow/tmpl/stages/default.php]: Perhaps we should add ```$this->escape(...)``` here, too?

Comment (brianteeman) [reply administrator/components/com_workflow/tmpl/stages/default.php]: i was following your code style above for title

Comment (brianteeman) [reply administrator/components/com_workflow/tmpl/stages/default.php]: @wilsonge please advise

Comment (brianteeman) [reply administrator/components/com_workflow/tmpl/stages/default.php]: fixed

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**